services/publicapi/waiter_test.go

```diff
@@ -190,7 +190,8 @@ func TestPublicApiWaiter_WaitGracefulShutdown(t *testing.T) {
 		wc2 := waiter.add(key)
 
 		var waitTillCancelled = func(wc *waiterChannel) {
-			ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
+			var cancel func()
+			ctx, cancel = context.WithTimeout(ctx, 1*time.Second)
```

Comment:
why?

Reply:
overshadow  ... do you prefer the internal ctx will have a diff name ?

Reply:
changed name

Reply:
yes